src/kokkos/test/kokkos/VertexFinder_t.cc

```diff
@@ -0,0 +1,299 @@
+
+  Event ev;
+
+  float eps = 0.1f;
+  std::array<float, 3> par{{eps, 0.01f, 9.0f}};
+  for (int nav = 30; nav < 80; nav += 20) {
+    ClusterGenerator gen(nav, 10);
+
+    for (int i = 8; i < 20; ++i) {
+      auto kk = i / 4;  // M param
+
+      gen(ev);
+
+      Kokkos::parallel_for(
+          "init", team_policy(1, 1), KOKKOS_LAMBDA(const member_type& team_member) {
+            onGPU_d(0).nvFinal = 0;
+            ws_d(0).ntrks = 0;
+            ws_d(0).nvIntermediate = 0;
+          });
+
+      std::cout << "v,t size " << ev.zvert.size() << ' ' << ev.ztrack.size() << std::endl;
+      auto nt = ev.ztrack.size();
+
+      auto ws_h = Kokkos::create_mirror_view(ws_d);
+      auto onGPU_h = Kokkos::create_mirror_view(onGPU_d);
+
+      // copy data to host mirror
+      ::memcpy(LOC_WS(ntrks), &nt, sizeof(uint32_t));
+      ::memcpy(LOC_WS(zt), ev.ztrack.data(), sizeof(float) * ev.ztrack.size());
+      ::memcpy(LOC_WS(ezt2), ev.eztrack.data(), sizeof(float) * ev.eztrack.size());
+      ::memcpy(LOC_WS(ptt2), ev.pttrack.data(), sizeof(float) * ev.eztrack.size());
+      // deep copy from host to device
+      Kokkos::deep_copy(ws_d, ws_h);
+
+      std::cout << "M eps, pset " << kk << ' ' << eps << ' ' << (i % 4) << std::endl;
+
+      if ((i % 4) == 0)
+        par = {{eps, 0.02f, 12.0f}};
+      if ((i % 4) == 1)
+        par = {{eps, 0.02f, 9.0f}};
+      if ((i % 4) == 2)
+        par = {{eps, 0.01f, 9.0f}};
+      if ((i % 4) == 3)
+        par = {{0.7f * eps, 0.01f, 9.0f}};
+
+      uint32_t nv = 0;
+      Kokkos::parallel_for(
+          "print", team_policy(1, 1), KOKKOS_LAMBDA(const member_type& team_member) {
+            printf("nt,nv %d %d,%d\n", ws_d(0).ntrks, onGPU_d(0).nvFinal, ws_d(0).nvIntermediate);
+          });
+
+      Kokkos::fence();
+
+#ifdef ONE_KERNEL
+      team_policy policy = team_policy(1, Kokkos::AUTO()).set_scratch_size(0, Kokkos::PerTeam(1024));
+      Kokkos::parallel_for(
+          policy, KOKKOS_LAMBDA(const member_type& team_member) {
+            vertexFinderOneKernel(onGPU_d, ws_d, kk, par[0], par[1], par[2], team_member);
+          });
+#else
+      team_policy policy = team_policy(1, Kokkos::AUTO()).set_scratch_size(0, Kokkos::PerTeam(1024));
+      Kokkos::parallel_for(
+          policy, KOKKOS_LAMBDA(const member_type& team_member) {
+            CLUSTERIZE(onGPU_d, ws_d, kk, par[0], par[1], par[2], team_member);
+          });
+#endif
+      Kokkos::parallel_for(
+          "print", team_policy(1, 1), KOKKOS_LAMBDA(const member_type& team_member) {
+            printf("nt,nv %d %d,%d\n", ws_d(0).ntrks, onGPU_d(0).nvFinal, ws_d(0).nvIntermediate);
+          });
+
+      Kokkos::fence();
+
+      Kokkos::parallel_for(
+          policy,
+          KOKKOS_LAMBDA(const member_type& team_member) { fitVerticesKernel(onGPU_d, ws_d, 50.f, team_member); });
+      // deep copy from device to host
+      Kokkos::deep_copy(onGPU_h, onGPU_d);
```

Comment:
```suggestion
      Kokkos::deep_copy(KokkosExecSpace(), onGPU_h, onGPU_d);
```
and very likely it would be good to add `KokkosExecSpace().fence()` after it (just in case we ever try to use asynchronous execution here).